ext/opcache/zend_persist.c

```diff
@@ -190,6 +190,9 @@ static zend_ast *zend_persist_ast(zend_ast *ast)
 		node = (zend_ast *) copy;
 	} else if (ast->kind == ZEND_AST_OP_ARRAY) {
 		zend_ast_op_array *copy = zend_shared_memdup(ast, sizeof(zend_ast_op_array));
+		/* We're holding a separate reference to the op_array in the AST. Release it
+		 * early because zend_persist_op_array() is destructive. */
+		destroy_op_array(copy->op_array);
```

Comment:
Looking at this, it looks like a similar issue I ran into [in my little rfc implementation](https://github.com/php/php-src/pull/17895/files#diff-4f708cc9a2a26775241c72cc9926d4c18018ae51ebd35a34b54a31d59a6234f9) where the pointer is lost when rewritten. So, I added an extra pass to rewrite those pointers specifically. I stumbled upon your PR while looking for prior art, so maybe it is a similar problem? It sounds similar but I'm not familiar enough with how attributes work (yet).